src/GafferOSL/OSLObject.cpp

```diff
@@ -88,21 +120,43 @@ const Gaffer::IntPlug *OSLObject::interpolationPlug() const
 	return getChild<IntPlug>( g_firstPlugIndex + 1 );
 }
 
+ScenePlug *OSLObject::resampledInPlug()
+{
+	return getChild<ScenePlug>( g_firstPlugIndex + 2 );
+}
+
+const ScenePlug *OSLObject::resampledInPlug() const
+{
+	return getChild<ScenePlug>( g_firstPlugIndex + 2 );
+}
+
+StringPlug *OSLObject::resampledNamesPlug()
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 3 );
+}
+
+const StringPlug *OSLObject::resampledNamesPlug() const
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 3 );
+}
+
 void OSLObject::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const
 {
 	SceneElementProcessor::affects( input, outputs );
 
-	if( input == shaderPlug() )
+	if( input == shaderPlug() ||
+		input == inPlug()->transformPlug() ||
+		input == interpolationPlug() ||
+		input == resampledInPlug()->objectPlug()
+		)
 	{
 		outputs.push_back( outPlug()->objectPlug() );
-	}
-	else if( input == inPlug()->transformPlug() )
-	{
+		outputs.push_back( outPlug()->objectPlug() );
```

Comment:
There seem to be three calls to `outputs.push_back( outPlug()->objectPlug() )` here - we just need the one, as in my example from the previous review.